qiskit/primitives/backend_estimator.py

```diff
@@ -57,11 +55,19 @@ def _run_circuits(
     for circ in circuits:
         metadata.append(circ.metadata)
         circ.metadata = {}
-
-    job = backend.run(circuits, **run_options)
-    if monitor:
-        job_monitor(job)
-    return job.result(), metadata
```

Comment:
Looks like the job monitoring was an oversight that never should have been included, since it's IBM-specific, I think?  Technically I guess it's a breaking change, since `monitor` could have been in the `run_options` that get unpacked into this function - is it worth a "bugfix" note to say it's removed?

Reply:
Yeah it was an oversight, it is IBM backend specific and I assumed it was ok to just remove it here because it was undocumented if you set `monitor` to something that evaluated true in the run options it would trigger this. But you're right we should document it as a fix, I'll add that to the release notes

Reply:
`monitor` option was originally for internal development use. I agree to remove it.